lib/response.js

```diff
@@ -80,9 +80,10 @@ module.exports = {
    */
 
   set status(code) {
+    if (this.headerSent) return;
+
     assert('number' == typeof code, 'status code must be a number');
     assert(statuses[code], `invalid status code: ${code}`);
-    assert(!this.res.headersSent, 'headers have already been sent');
```

Comment:
I understand my opinion wasn't asked for, but I'm considering if this could be a breaking change.
Is there any reason a user would rely on something like:

```js
try {
  ctx.status = 201
} catch (err) {
  if (err.message.match(/headers have already been sent/) {
    // ...
  }
}
```

Reply:
yes, it is hard to make this decision, but even in this situation, you can do nothing but log this error because the http header has already sent.